src/js/components/Baani.js

```diff
@@ -114,8 +114,32 @@ export default class Baani extends React.PureComponent {
     }
   };
 
+  makeSelection = e => {
+    const shareDivs = document.querySelectorAll('.share');
+    shareDivs.forEach(function(sd) {
+      sd.style.display = 'none';
+    });
+    if (window.getSelection().toString()) {
+      const baaniDiv = e.currentTarget;
+      var range = document.createRange();
+      range.selectNode(baaniDiv);
+      window.getSelection().removeAllRanges();
+      window.getSelection().addRange(range);
+      const shareDiv = baaniDiv.querySelector('.share');
+      shareDiv.style.display = 'block';
+    }
+  };
+
   componentDidMount() {
     this._scrollToHiglight();
+    const lineDivs = document.querySelectorAll('.line');
+    const ms = this.makeSelection;
+    lineDivs.forEach(function(ld) {
```

Comment:
Can we instead do it in the React side? Like event listeners on the line components

Reply:
Fixed @bogas04